lib/serverless-tools/deployer/function_config.rb

```diff
@@ -2,12 +2,31 @@
 
 module ServerlessTools
   module Deployer
-    FunctionConfig = Struct.new(:repo, :s3_archive_name, :handler_file, :bucket, :name, keyword_init: true) do
+    FunctionConfig = Struct.new(
+      :repo,
+      :s3_archive_name,
+      :handler_file,
+      :bucket,
+      :name,
+      :dockerfile,
+      keyword_init: true
+    ) do
       def runtime
         unless handler_file.nil?
-          return "ruby" if handler_file.split(".").last == "rb"
+          case file_extension(handler_file)
+          when "rb"
+            "ruby"
+          when "R"
```

Comment:
I wonder if instead of checking whether the file is an R file - we check is dockerfile is present? That way, if folks want to add another Docker project with a different runtime we don't need need to update serverless tools.